app/src/main/java/org/oppia/app/home/HomeActivity.kt

```diff
@@ -24,4 +28,17 @@ class HomeActivity : InjectableAppCompatActivity(), RouteToExplorationListener,
   override fun routeToTopic(topicId: String) {
     startActivity(TopicActivity.createTopicActivityIntent(this, topicId))
   }
+
+  override fun onBackPressed() {
+    AlertDialog.Builder(this, R.style.AlertDialogTheme)
+      .setMessage(R.string.home_activity_back_dialog_message)
+      .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
+        dialog.dismiss()
+      }
+      .setPositiveButton(R.string.home_activity_back_dialog_exit) { _, _ ->
+        val intent = Intent(this, ProfileActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
```

Comment:
Why not finish the activity, instead? Won't the profile activity always be present at the bottom of the stack or do we want to explicitly end it? I'm a bit confused by the activity navigation flows here.

Reply:
I used an intent because of the live data bug. If the profiles were edited in the settings menu and we only called finish, the ProfileActivity does not get the update.

Reply:
Ah, please document that here. It wasn't clear, and if someone changes this in the future they may regress that.

Reply:
Updated with comment.